### PR TITLE
Automate removal of CMAction, cmaction, LineageAction, KeyHandler and…

### PR DESCRIPTION
**Pull request by Herrie82**
… Doze

CMAction, LineageAction, KeyHandler and Doze all use SettingsLib which
we don't provide, so delete these folders completely so they don't get
build and we need less forking and patching of device repos.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

### CODE REVIEW

**Comment by Lyuu17** [comment]
In some devices it's "MotoActions"

**Comment by Herrie82** [comment]
Lyudmila17: I couldn't find any references for MotoActions on LineageOS GitHub code.

Maybe you mean MotoDoze? I checked the various Doze variants, seems they're all in the doze folder, regardless if they're called MotoDoze, XiaomiDoze, ZteDoze etc ;) Removing the doze folder seems to work for me to get things building on Athene for example.

**Comment by Lyuu17** [comment]
I meant MotoActions. Seems to be on another repos than LineageOS ones.

https://github.com/DirtyUnicorns/android_device_motorola_clark
https://github.com/AospExtended-Devices/device_motorola_potter
https://github.com/PixelExperience-Devices/device_motorola_cedric

**Comment by Herrie82** [comment]
@Lyudmila17 We cannot cover all possible scenarios in the world. I think we should focus on LineageOS repos. In the ones you quoted above, only one is 14.1 based that has MotoActions, the other two are 15.1 based.

File: setup [hunk 1]
```diff
@@ -68,6 +76,18 @@ else
         sed -r 's/(,context=.*:s0)//' $j >$j".tmp" && mv $j".tmp" $j
     done
 
+    # Since we don't have SettingsLib, remove components that rely on it (which we therefore also don't use)
+    # such as CMActions/LineageActions, KeyHandler and Doze. Simply removing the folder will disable them.
+    cd $DEVICE_TREE
+    echo "I: Removing components relying on SettingsLib from: device/$VENDOR/$DEVICE"
+    rm -rf cmactions
```
Comment (vanyasem) [inline setup]: I don't like this being duplicated 4 times. If I was to add a folder, I would certainly forget to add it to one of the "rm"s. You should maybe define a function to delete files. Also, you could probably use `find . -iname "file" -exec rm {} \;` OR `rm -rf [Dd]oze` syntax to ignore case

Comment (Herrie82) [reply setup]: Fair point there, I'll try and see if I can rework this shortly. Been busy moving, so have little time for dev projects at the moment, but will see later this week.

**Comment by Herrie82** [comment]
@vanyasem Thanks for the pointers, updated the PR a bit, tested it for a few devices and seems to do what's needed :)

File: setup [hunk 2]
```diff
@@ -1,5 +1,12 @@
 #!/bin/bash
 
+function removeUnneededFolders {
+rm -rf [Cc][Mm]actions
```
Comment (jbruechert) [inline setup]: The contents of the function should be indented by 4 spaces. (Maybe it's just displayed incorrectly on my phone, sorry then)

**Comment by Herrie82** [comment]
@JBBgameich Sorry it was very early this morning when I pushed it, fixed now :)

**Pull request merged**